Week-1/assignment1_1.md

```diff
@@ -0,0 +1,107 @@
+
+
+
+UI Backend: This component uses the user interface methods of the underlying operating system. It is mainly used for drawing basic widgets (windows and combo boxes).
+Data Storage/Persistence: It is a persistent layer. A web browser needs to store various types of data locally, for example, cookies. As a result, browsers must be compatible with data storage mechanisms such as WebSQL, IndexedDB, FileSystem, etc.
+
+7-
+BLINK render engine
+GECO render engine
+
+Rendering engine and its use.
+The responsibility of the rendering engine is well… Rendering, that is display of the requested contents on the browser screen.
+Different browsers use different rendering engines: Internet Explorer uses Trident, Firefox uses Gecko, Safari uses WebKit. Chrome and Opera (from version 15) use Blink, a fork of WebKit.
+
+uses of render engine:
+The rendering engine will start getting the contents of the requested document from the networking layer. This will usually be done in 8kB chunks.
+
+
+Parsing:
+The rendering engine will start parsing the HTML document and convert elements to DOM nodes in a tree called the "content tree". The engine will parse the style data, both in external CSS files and in style elements. Styling information together with visual instructions in the HTML will be used to create another tree: the render tree.
+Parsing a document means translating it to a structure the code can use. The result of parsing is usually a tree of nodes that represent the structure of the document. This is called a parse tree or a syntax tree.
+
+HTML Parser #
+The job of the HTML parser is to parse the HTML markup into a parse tree.
+
+WebKit CSS parser #
+WebKit uses Flex and Bison parser generators to create parsers automatically from the CSS grammar files. As you recall from the parser introduction, Bison creates a bottom up shift-reduce parser. Firefox uses a top down parser written manually. In both cases each CSS file is parsed into a StyleSheet object. Each object contains CSS rules. The CSS rule objects contain selector and declaration objects and other objects corresponding to CSS grammar.
+
+
+Render tree construction #
+While the DOM tree is being constructed, the browser constructs another tree, the render tree. This tree is of visual elements in the order in which they will be displayed. It is the visual representation of the document. The purpose of this tree is to enable painting the contents in their correct order.
+
+Firefox calls the elements in the render tree "frames". WebKit uses the term renderer or render object.
+
+A render knows how to lay out and paint itself and its children.
+
+
+
+Order of script processing
+
+
+Layout and Painting
+imultaneously, the browser also creates a render tree. This tree includes both the styling information as well as the visual instructions that define the order in which the elements will be displayed. The render tree ensures that the content is displayed in the desired order.
```

Comment:
imultaneously?

Reply:
wrong type mam ,It is simultaneously .I updated it mam